File: dev-docs/snippets/document-blob/read.ts
import type { JsBaoClient } from "js-bao-wss-client";

export async function read(
  client: JsBaoClient,
  documentId: string,
  blobId: string,
) {
  // #region example
  const blobs = client.document(documentId).blobs();
  // `as` picks the return format: "uint8array" | "arrayBuffer" | "blob" | "text".
  const bytes = await blobs.read(blobId, { as: "arrayBuffer" });
  const text = await blobs.read(blobId, { as: "text", forceRedownload: true });
  // #endregion example
  return { bytes, text };
}

File: dev-docs/snippets/document-blob/read.swift
import Foundation
import JsBaoClient

func read(
  client: JsBaoClient,
  documentId: String,
  blobId: String
) async throws {
  // #region example
  let blobs = client.documents.blobs(documentId: documentId)
  // Base read returns raw `Data`. Pass `force: true` to bypass the cache.
  let bytes: Data = try await blobs.read(blobId: blobId)
  let text = String(data: bytes, encoding: .utf8)
  // #endregion example
  _ = (bytes, text)
}
